### PR TITLE
enable cilium to retrieve k8s node node cidr

### PR DESCRIPTION
**Pull request by aanm**
Signed-off-by: André Martins <andre@cilium.io>

### CODE REVIEW

File: daemon/main.go [hunk 1]
```diff
@@ -348,6 +348,7 @@ func configDaemon(ctx *cli.Context) {
 }
 
 func initEnv(ctx *cli.Context) error {
+
```
Comment (tgraf) [inline daemon/main.go]: unwanted

File: common/const.go [hunk 2]
```diff
@@ -71,6 +72,17 @@ const (
...
+	LastFreeIPv4PrefixKeyPath = OperationalPath + "/NodeIPv4Prefixes/LastIPv4Prefix"
+	// NodeAddrKeyPath is the base path where the Node Addresses are stored in the KVStore.
+	NodeAddrKeyPath = OperationalPath + "/NodeIPv4Prefixes/NodeAddrs"
+	// IPv4PrefixKeyPath is the base path where IPv4 prefix is stored in the KVStore.
+	IPv4PrefixKeyPath = OperationalPath + "/NodeIPv4Prefixes/IPv4Prefixes"
+	// MaxSetOfIPv4Prefix is maximum number of set of service IDs that can be stored in consul.
+	MaxSetOfIPv4Prefix = uint32(0xFF)
+	// FirstFreeServiceID is the first ID for which the services should be assigned.
+	FirstFreeIPv4Prefix = uint32(0)
+
```
Comment (tgraf) [inline common/const.go]: looks like added by mistake

**Review by tgraf — CHANGES_REQUESTED** [review]
Looks great. I think this should be the default as it requires no additional configuration or cluster state sync at all.

File: daemon/daemon/daemon.go [hunk 3]
```diff
@@ -367,8 +402,21 @@ func NewDaemon(c *Config) (*Daemon, error) {
 		}
 	}
 
-	if err := d.init(); err != nil {
-		log.Fatalf("Error while initializing daemon: %s\n", err)
+	if os.Getenv(common.K8sEnvNodeNameSpec) != "" {
```
Comment (tgraf) [inline daemon/daemon/daemon.go]:
```
if node_name := os.Getenv(..); node_name != "" {
    if err := d.useK8sNodeCIDR(node_name); err != nil {
```

**Review by tgraf — APPROVED** [review]
One minor nit, rest looks good.

File: daemon/daemon/daemon.go (hunk 3 above)
Comment (aanm) [reply daemon/daemon/daemon.go]: Ah yes, I missed that one. Thanks

**Pull request merged**